src/extensions/jana/JOmniFactory.h

```diff
@@ -259,6 +267,39 @@ class JOmniFactory : public JMultifactory {
         m_parameters.push_back(parameter);
     }
 
+    template <typename AlgorithmsT>
+        requires UsesAlgorithmsProperties<AlgorithmsT, ConfigT>
+    void RegisterAllParameters(std::unique_ptr<AlgorithmsT>& algo, std::vector<std::unique_ptr<ParameterBase>>& parameters) {
+        for (const auto& [key, prop] : algo->getProperties()) {
+            std::visit(
+                [this, key = key, &parameters](auto&& val) {
+                    using T = std::decay_t<decltype(val)>;
+                    if constexpr (std::is_fundamental_v<T> || std::is_same_v<T, std::string>) {
```

Comment:
This is incomplete because JParameterManager can parse the algorithms::Property of type `T = std::vector<U>` where `U` is a fundamental type or std::string. However, we can't use
```suggestion
                    if constexpr (std::is_fundamental_v<T> || std::is_same_v<T, std::string>
                        || std::is_same_v<std::vector<typename T::value_type, T> && std::is_fundamental_v<typename T::value_type>) {
```
because it tries to apply this to the fundamental types too, and fails due to `T::value_type` not existing for e.g. `T = bool`. I'm not sure yet how to solve this.

Reply:
@nathanwbrei @veprbl Maybe you have some insights on how to structure a type test here.